src/logic/currencyValues/store/selectors/index.ts

```diff
@@ -1,10 +1,14 @@
-import { List } from 'immutable'
+import { List, Map, RecordOf } from 'immutable'
 import { createSelector } from 'reselect'
 
 import { CURRENCY_VALUES_KEY } from 'src/logic/currencyValues/store/reducer/currencyValues'
 import { safeParamAddressFromStateSelector } from 'src/routes/safe/store/selectors'
+import { AppReduxState } from '../../../../store'
+import { CurrencyRateValue } from '../model/currencyValues'
+import { BigNumber } from 'bignumber.js'
 
-export const currencyValuesSelector = (state) => state[CURRENCY_VALUES_KEY]
+export const currencyValuesSelector = (state: AppReduxState): Map<string, RecordOf<CurrencyRateValue>> =>
```

Comment:
let's extract the return type to a separate type, place it close to the state definition and import it from there. we're possibly going to re-use it

Reply:
also RecordOf type could be extracted and put to the same file as model definition